### ISSUE TITLE
Specifying a directory in staticpaths shouldn't migrate files one by one

### ISSUE THREAD

**Issue by alex-way** [staticjinja/staticjinja#166]
When specifying a directory in `staticpaths`, staticjinja should copy the entire directory, and all sub files in one operation. Currently the files in the directory are copied one by one, which for a large number of files can be extremely slow.

**Comment by NickCrews** [comment]
This is true. Can you be more specific as to "extremely slow"? How long for how many files?

Doing them all at once would require a larger re-structuring, it is nice how right now the code only operates on a template-by-template basis, and we don;t have to worry about bulk operations.

Perhaps another solution would be to use async/await, I believe that copying files would be a good application for that, and it might need less refactoring.
